infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -97,25 +101,92 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, LogReplicationContext
         this.runtime = runtime;
         this.corfuStore = new CorfuStore(runtime);
         this.replicationContext = replicationContext;
+        this.metadataTable = tryOpenMetadataTable(this.corfuStore);
+        this.statusTable = tryOpenReplicationStatusTable(this.corfuStore);
+        this.replicationEventTable = tryOpenReplicationEventTable(this.corfuStore);
+    }
 
+    private static Table<LogReplicationSession, ReplicationMetadata, Message>
+    tryOpenMetadataTable(CorfuStore corfuStore) {
+        Table<LogReplicationSession, ReplicationMetadata, Message> table = null;
         try {
-            this.metadataTable = this.corfuStore.openTable(NAMESPACE, METADATA_TABLE_NAME,
+            table = corfuStore.openTable(NAMESPACE, METADATA_TABLE_NAME,
                     LogReplicationSession.class, ReplicationMetadata.class, null,
                     TableOptions.fromProtoSchema(ReplicationMetadata.class));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening replication metadata table", ee);
+            throw new ReplicationWriterException(ee);
+        }
+        return table;
+    }
 
-            this.statusTable = this.corfuStore.openTable(NAMESPACE, LogReplicationUtils.REPLICATION_STATUS_TABLE_NAME,
+    private static Table<LogReplicationSession, ReplicationStatus, Message>
+    tryOpenReplicationStatusTable(CorfuStore corfuStore) {
+        Table<LogReplicationSession, ReplicationStatus, Message> table = null;
+        try {
+            table = corfuStore.openTable(NAMESPACE, LogReplicationUtils.REPLICATION_STATUS_TABLE_NAME,
                     LogReplicationSession.class, ReplicationStatus.class, null,
-                    TableOptions.fromProtoSchema(ReplicationStatus.class));
+                    TableOptions.fromProtoSchema(ReplicationMetadata.class));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening replication status table", ee);
+            throw new ReplicationWriterException(ee);
+        }
+        return table;
+    }
 
-            this.replicationEventTable = this.corfuStore.openTable(NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
+    private static Table<ReplicationEventInfoKey, ReplicationEvent, Message>
+    tryOpenReplicationEventTable(CorfuStore corfuStore) {
```

Comment:
nit: we could have a generic `Table<K, V, M> tryOpenTable()` to avoid some reuse of code

Reply:
done Shreay, wasn't able to do a way without warnings on use of generic types but I think this way reduces code duplication. Could you please let me know if this is ok? thanks

Reply:
looks good!